src/kili/queries/asset/helpers.py

```diff
@@ -73,30 +107,75 @@ def download_file(url: str, external_id: str, local_dir_path: Path) -> Path:
 
 def assert_required_fields_existence(asset: dict) -> None:
     """Get a list of an asset field from a generator of kili assets"""
-    required_fields = ["content", "externalId"]
+    required_fields = ["content", "externalId", "jsonContent", "project.inputType"]
     for field in required_fields:
         if field not in asset.keys():
             raise MissingPropertyError(
                 f"The asset does not have the {field} field. Please add it to the fields when"
-                " querying assets "
-            )
-        content = asset["content"]
-        if content == "":
-            raise NotImplementedError(
-                "The media of the asset is not stored in the content field. If your asset is video"
-                " imported from frames or a richText asset, the media is stored in the jsonContent"
-                " field. The download in local of such asset media is currently not supported."
+                " querying assets."
             )
 
 
+# pylint: disable=too-many-branches
 def download_asset_content(assets: List[dict], local_dir_path: Path) -> List[dict]:
```

Comment:
the purpose of this function was to download content (see the function name). I would find it clearer to have two functions download_asset_content and download_asset_frames and to call the right one. (And see where to put rich text)